misc/v3-task-endpoint-validator/v3-task-endpoint-validator.go

```diff
@@ -0,0 +1,475 @@
+		return err
+	}
+
+	if err = verifyNetworksResponse(containerMetadataResponseMap["Networks"]); err != nil {
+		return err
+	}
+
+	if err = isFieldEmpty(containerMetadataResponseMap, "Health"); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func verifyLimitResponse(limitRawMsg json.RawMessage) error {
+	var err error
+	limitResponseMap := make(map[string]json.RawMessage)
+	json.Unmarshal(limitRawMsg, &limitResponseMap)
+
+	if err = isFieldEqual(limitResponseMap, "CPU", float64(0)); err != nil {
+		return err
+	}
+
+	if err = isFieldEqual(limitResponseMap, "Memory", float64(50)); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func verifyNetworksResponse(networksRawMsg json.RawMessage) error {
+	var err error
+	if networksRawMsg == nil {
+		notEmptyErrMsg("Networks")
+	}
+
+	var networksResponseArray []json.RawMessage
+	json.Unmarshal(networksRawMsg, &networksResponseArray)
+
+	if len(networksResponseArray) == 1 {
+		networkResponseMap := make(map[string]json.RawMessage)
+		json.Unmarshal(networksResponseArray[0], &networkResponseMap)
+
+		if err = isFieldEqual(networkResponseMap, "NetworkMode", "awsvpc"); err != nil {
+			return err
+		}
+
+		if err = isFieldEmpty(networkResponseMap, "IPv4Addresses"); err != nil {
+			return err
+		}
+
+		var ipv4AddressesResponseArray []json.RawMessage
+		json.Unmarshal(networkResponseMap["IPv4Addresses"], &ipv4AddressesResponseArray)
+
+		if len(ipv4AddressesResponseArray) != 1 {
+			return fmt.Errorf("incorrect number of IPv4Addresses, expected 1, received %d",
+				len(ipv4AddressesResponseArray))
+		}
+
+		isAWSVPCNetworkMode = true
+	} else if len(networksResponseArray) != 0 {
+		return fmt.Errorf("incorrect number of networks, expected 0 or 1, received %d",
+			len(networksResponseArray))
+	}
+
+	return nil
+}
+
+func metadataResponse(client *http.Client, endpoint string) ([]byte, error) {
+	var resp []byte
+	var err error
+	for i := 0; i < maxRetries; i++ {
+		resp, err = metadataResponseOnce(client, endpoint)
+		if err == nil {
+			return resp, nil
+		}
+		fmt.Fprintf(os.Stderr, "Attempt [%d/%d]: unable to get metadata response from '%s': %v",
+			i, maxRetries, endpoint, err)
+		time.Sleep(durationBetweenRetries)
+	}
+
+	return nil, err
+}
+
+func metadataResponseOnce(client *http.Client, endpoint string) ([]byte, error) {
+	resp, err := client.Get(endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get response: %v", err)
+	}
+	if resp.Body != nil {
+		defer resp.Body.Close()
+	}
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("incorrect status code  %d", resp.StatusCode)
+	}
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read response body: %v", err)
+	}
+
+	return body, nil
+}
+
+func notEmptyErrMsg(fieldName string) error {
+	return fmt.Errorf("field %s should not be empty", fieldName)
+}
+
+func isFieldEmpty(rawMsgMap map[string]json.RawMessage, fieldName string) error {
```

Comment:
nit: isFieldEmpty and isFieldEqual should probably not start with 'is' since there are not returning bool. and 'fieldEmpty' should probably be 'fieldNotEmpty'